pkg/trace/metrics/client.go

```diff
@@ -6,60 +6,12 @@
 package metrics
 
 import (
-	"time"
-
 	"github.com/DataDog/datadog-go/statsd"
 )
 
 // StatsClient represents a client capable of sending stats to some stat endpoint.
-type StatsClient interface {
-	Gauge(name string, value float64, tags []string, rate float64) error
-	Count(name string, value int64, tags []string, rate float64) error
-	Histogram(name string, value float64, tags []string, rate float64) error
-	Timing(name string, value time.Duration, tags []string, rate float64) error
-	Flush() error
-}
+type StatsClient = statsd.ClientInterface
 
 // Client is a global Statsd client. When a client is configured via Configure,
 // that becomes the new global Statsd client in the package.
 var Client StatsClient = (*statsd.Client)(nil)
-
-// Gauge calls Gauge on the global Client, if set.
```

Comment:
This is already done by the main client: https://github.com/DataDog/datadog-go/blob/master/statsd/statsd.go#L463

Every method checks for `nil` so the whole `StatsClient` is useless. I created a type alias line 13 to not change to much code. I'll let @DataDog/agent-apm update the rest of the trace agent to replace `metrics.StatsClient` for `statsd.ClientInterface` if they wan to.

Reply:
Agreed. All this is fine.

Reply:
I'll leave it in the end since it's changing to much code on trace side